src/plugins/intel_gpu/src/runtime/format.cpp

```diff
@@ -344,4 +344,30 @@ const std::vector<std::pair<size_t, int>> format::per_axis_block_size(format fmt
 
     return sizes_for_dims;
 }
+
+format format::find_format(const std::vector<uint64_t>& order,
+                           const std::vector<std::pair<size_t, int>>& block_sizes,
+                           bool is_weights,
+                           bool is_grouped,
+                           bool is_image_2d,
+                           bool is_winograd,
+                           bool is_nv12) {
+    auto filter = [&](const std::pair<cldnn::format::type, cldnn::format_traits>& trait) -> bool {
+        return trait.second._order == order &&
+               trait.second.block_sizes == block_sizes &&
+               format::is_weights_format(trait.first) == is_weights &&
+               format::is_grouped(trait.first) == is_grouped &&
+               format::is_image_2d(trait.first) == is_image_2d &&
+               format::is_winograd(trait.first) == is_winograd &&
+               format::is_nv12(trait.first) == is_nv12;
+    };
+    const auto& finded_format = std::find_if(format_traits_map.begin(), format_traits_map.end(), filter);
+    OPENVINO_ASSERT(finded_format != format_traits_map.end(), "[GPU] Cannot find a format with the specified parameters");
+
+    const auto& finded_format_from_end = std::find_if(format_traits_map.rbegin(), format_traits_map.rend(), filter);
+    OPENVINO_ASSERT(finded_format->first == finded_format_from_end->first, "[GPU] Cannot find a format. Specified parameters are ambiguous");
```

Comment:
Wouldn't it be better to iterate over format_traits_map manually only once and count the number of occurrences?